sdk/android/api/org/webrtc/HardwareVideoEncoderFactory.java

```diff
@@ -27,6 +27,11 @@
 public class HardwareVideoEncoderFactory implements VideoEncoderFactory {
   private static final String TAG = "HardwareVideoEncoderFactory";
 
+  // We don't need periodic keyframes. But some HW encoders, Exynos in particular, fails to
+  // initialize with value -1 which should disable periodic keyframes according to the spec. Set it
+  // to 1 hour.
+  private static final int PERIODIC_KEY_FRAME_INTERVAL_S = 3600;
```

Comment:
Could this be why so many Exynos devices need to be forced to use software encoders?

And if so, is one hour long enough? Why not MAX_INT?

Reply:
Do you mean in our blacklist?

Reply:
As for 1 hour value:  for almost all calls 1 hours is effectively the same as MAX_INT, and maybe this was a safer value because maybe some hardware codecs blow up with a larger value.

Reply:
Yes. But now that I am reading this comment it is about initialization, those other issues were usually at random times during calls. Resolving anyway.